src/rollup/utils/persist/expiring-local-storage.js

```diff
@@ -0,0 +1,53 @@
+var PAYLOAD_KEY = '$';
+var EXPIRES_KEY = '@';
+
+function safeJsonParse (json) {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    // recover
+  }
+}
+
+export default {
+  getItem: function (key) {
+    var value = safeJsonParse(localStorage.getItem(key));
+
+    if (value && value[EXPIRES_KEY]) {
+      if (Date.parse(value[EXPIRES_KEY]) < Date.now()) {
+        localStorage.removeItem(key);
+        return null;
+      }
+      return value[PAYLOAD_KEY] || value;
+    }
+    return localStorage.getItem(key);
+  },
+
+  setItem: function (key, payload, expiresOn) {
+    if (!expiresOn) {
+      expiresOn = new Date();
+      expiresOn.setDate(expiresOn.getDate() + 365);
```

Comment:
do we want to expire in a year or 30 days? thought we talked about being more aggressive on the window but perhaps its a smaller problem when in localStorage. probably ok to leave as is for now so ist 1 to 1 with whats there then can change in a follow up PR if we so choose.

Reply:
i just went with 1 year because that's what it was with cookies. We talked about extending the configuration to make this configurable

Reply:
works for me. i think having this at a year also puts much less emphasis on some of the weirdness around extending expiration dates as well (given we have a year to solve it and at that point its unlikely that the interaction from a year previous is relevant anyway).

Reply:
:thinking: In our conversation, I thought we landed on 30 days? Also I am less concerned about it, cause every time we read it it gets extended by 30 days, right?